File: 2021/tests/exercises/test_exercise_9.py
from adventofcode.exercises import exercise_9
from adventofcode import utils

TEST_DATA = """2199943210
3987894921
9856789892
8767896789
9899965678
"""


def test_example_on_exercise_a():
    assert exercise_9.run_a(TEST_DATA) == 15


def test_returns_correct_answer_on_exercise_a():
    input_data = utils.load_data(exercise_9.EXERCISE)
    assert exercise_9.run_a(input_data) == 448


def test_example_on_exercise_b():
    assert exercise_9.run_b(TEST_DATA) == 1134


def test_returns_correct_answer_on_exercise_b():
    input_data = utils.load_data(exercise_9.EXERCISE)
    assert exercise_9.run_b(input_data) == 1417248

File: 2024/adventofcode/tests/test_exercise_9.rs
use adventofcode::exercises::exercise_9;
use adventofcode::exercises::utils;

const TEST_DATA: &str = "2333133121414131402";

#[test]
fn test_a_returns_correct_answer_for_test_data() {
    assert_eq!(exercise_9::run_a(TEST_DATA), 1928);
}

#[test]
fn test_a_returns_correct_answer_on_exercise_a() {
    let input_data = utils::load_data(exercise_9::EXERCISE);
    assert_eq!(exercise_9::run_a(&input_data), 6334655979668);
}

#[test]
fn test_b_returns_correct_answer_for_test_data() {
    assert_eq!(exercise_9::run_b(TEST_DATA), 2858);
}

#[test]
fn test_b_returns_correct_answer_on_exercise_b() {
    let input_data = utils::load_data(exercise_9::EXERCISE);
    assert_eq!(exercise_9::run_b(&input_data), 6349492251099);
}
